README.md

```diff
@@ -1,170 +1,268 @@
-}
+If `featureCollection` parameter is set, this `lineString` parameter is ignored.
 
-DirectionsThemeResponse {
-  coordinates: number[][];
-  waypoints: Waypoints | undefined;
-}
+#### `themeSelectionType` - ThemeSelectionType | undefined
 
-Waypoints {
-  departure: number[];
-  arrival: number[];
-}
+```ts
+type ThemeSelectionType = "select" | "radioList";
 ```
 
-getPathByCoordinates function return a object of type DirectionsThemeResponse with the coordinates between two points and waypoints, if necessary, to create phantomJunction lines between waypoints and points
+Determines the UI for theme selection. Default set to `radioList`.
 
-- useRightClickToHandleActionPanel - `boolean | undefined` determine if the user should use the left or right click to handle the action panel
-- themeSelectionType - `"select" | "radioList"` UI to choose the theme. Default value is `"radioList"`
+#### `translate` - Function | undefined;
 
-### METHODS
+A function to provide locales. The default language is English (See [src/i18n.js](https://github.com/makinacorpus/mapbox-gl-path/blob/master/src/i18n.ts)).
 
-#### setLoopTrail
+#### `useRightClickToHandleActionPanel` - boolean | undefined;
 
-This is not applied if the number of points is less than 3.
+Boolean to use right or left mouse click to open the action panel. Default value is `false` (so left click).
 
-#### setOneWayTrail
+### Methods
 
 #### clearFeatureCollection
 
+Clears the paths from the map.
+
 #### getFeatureCollection
 
-return `GeoJSON.FeatureCollection<GeoJSON.Geometry>`
+Get the current FeatureCollection drawn.
+Returns `GeoJSON.FeatureCollection<GeoJSON.Geometry>`
 
 #### getLineString
 
-return `Feature<LineString>`
+Get the current drawn FeatureCollection drawn and concatenates the collection as a LineString. The `properties` will contain `path` and `point` element to describe to `Mapbox-gl-path` how to reconstruct the feature collection.
+Returns `Feature<LineString>`
 
 #### setFeatureCollection
 
-##### Parameter
+Parameter: `GeoJSON.FeatureCollection<GeoJSON.Geometry>`
 
-featureCollection - `GeoJSON.FeatureCollection<GeoJSON.Geometry>`
+The `featureCollection` includes features that could be of type `Point` or `Linestring`.  
+A `Linestring` can be a line between points or a phantom junction line (the latter must have a `isPhantomJunction` properties equals to `true`).
 
 #### setLineString
 
-##### Parameter
+Parameter: `GeoJSON.Feature<LineString>`
 
-lineString - `Feature<LineString>`
+It draws the lineString on the map.  
+If the lineString does not contains `path` and `point` properties to describe to `Mapbox-gl-path` how to build a feature collection, automatic Points will be added :
 
-## DEVELOPMENT
+- If the first and last point have the same coordinate, the lineString is considered to be looped so one point will be placed at that coordinate and two others will be placed at equal distance from each other.
+- Otherwise, two points should be added to the edges.
```

Comment:
```suggestion
- Otherwise, two points will be added to the edges.
```
To be consistent with this review https://github.com/makinacorpus/mapbox-gl-path/pull/30/files#r642286074